factory.rb

```diff
@@ -0,0 +1,45 @@
+require './app'
+require './processors'
+
+class Factory < UserPrompt
+  def initialize
+    super()
+    @app = App.new
+    @list_processor = ListProcessor.new(@app)
+    @create_processor = CreateProcessor.new(@app)
+  end
+
+  def run
+    show_allowed_options
+    prompt
+  end
+
+  def process(choice)
+    if (1..4).include?(choice)
+      @create_processor.process(choice)
+    else
+      @list_processor.process(choice)
+    end
+  end
+
+  def separator(char = '-', size = 100)
+    puts char * size
+  end
+
+  private
+
+  def show_allowed_options
+    options = [
+      'Create person',
+      'Create classroom',
+      'Create book',
+      'Create rental',
+      'List all books',
+      'List all people',
+      'List all classrooms',
+      'List all rentals by person',
+      'Quit'
```

Comment:
- [ ] I'm having some problems with what to create first. Also, you are not required for creating a classroom here. But it's a good idea. But when I try to create the person and see it, it's breaking the app. Kindly watch this video and fix the error. Make sure there are no errors while we run your app. 

https://user-images.githubusercontent.com/69011963/189472656-8ffee115-f885-4bd4-ac95-ebb7e6e9e9c7.mov